cmd/gh/main_test.go

```diff
@@ -76,3 +88,128 @@ check your internet connection or https://githubstatus.com
+		forkPath := "/repos/owner/repo/forks"
+		switch {
+		case path == forkPath:
+			forkResult := fmt.Sprintf(`{
+				"node_id": "123",
+				"name": "repo",
+				"clone_url": "https://github.com/someone/repo.git",
+				"created_at": "%s",
+				"owner": {
+					"login": "someone"
+				}
+			}`, time.Now().Format(time.RFC3339))
+			_, _ = w.Write([]byte(forkResult))
+		case gitPath.MatchString(path):
+			w.WriteHeader(http.StatusOK)
+		default:
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}))
+	t.Cleanup(s.Close)
+	os.Setenv("HTTP_PROXY", s.URL)
+	t.Cleanup(func() { os.Unsetenv("HTTP_PROXY") })
+
+	// Set up config directory and config files
+	os.Setenv("GH_CONFIG_DIR", tempDir)
+	t.Cleanup(func() { os.Unsetenv("GH_CONFIG_DIR") })
+	hosts := heredoc.Doc(`
+    github.localhost:
+      user: monalisa
+      oauth_token: TOKEN
+      git_protocol: https
+	`)
+	err := ioutil.WriteFile("hosts.yml", []byte(hosts), 0771)
+	assert.NoError(t, err)
+
+	// Set up local git repo
+	gitExe, err := safeexec.LookPath("git")
+	assert.NoError(t, err)
+	cmd := exec.Command(gitExe, "init", "--quiet", "repo")
+	err = cmd.Run()
+	assert.NoError(t, err)
+	assert.NoError(t, os.Chdir("repo"))
+	url := fmt.Sprintf("%s/owner/repo.git", "http://github.localhost")
+	cmd2 := exec.Command(gitExe, "remote", "add", "origin", url)
+	err = cmd2.Run()
+	assert.NoError(t, err)
+	// Add remote as resolved to skip graphql resolution
+	cmd3 := exec.Command(gitExe, "config", "--add", "remote.origin.gh-resolved", "base")
+	err = cmd3.Run()
+	assert.NoError(t, err)
+	// Set up server as git proxy
+	cmd4 := exec.Command(gitExe, "config", "http.http://github.localhost.proxy", s.URL)
+	err = cmd4.Run()
+	assert.NoError(t, err)
+
+	// Skip checking for updates
+	os.Setenv("GH_NO_UPDATE_NOTIFIER", "1")
+	t.Cleanup(func() { os.Unsetenv("GH_NO_UPDATE_NOTIFIER") })
+
+	// Skip terminal colors
+	os.Setenv("NO_COLOR", "1")
+	t.Cleanup(func() { os.Unsetenv("NO_COLOR") })
+	os.Setenv("CLICOLOR", "0")
+	t.Cleanup(func() { os.Unsetenv("CLICOLOR") })
+
+	// Set integration test environment variable
+	os.Setenv("GH_INTEGRATION_TEST", "1")
+	t.Cleanup(func() { os.Unsetenv("GH_INTEGRATION_TEST") })
+
+	// Set up command and IO
+	origArgs := os.Args
+	origOut := os.Stdout
+	origErr := os.Stderr
+	t.Cleanup(func() {
+		os.Args = origArgs
+		os.Stdout = origOut
+		os.Stderr = origErr
+	})
+
+	buf := &bytes.Buffer{}
+	c, _, err := vt10x.NewVT10XConsole(expect.WithStdout(buf), expect.WithDefaultTimeout(time.Second))
+	if err != nil {
+		t.Error(err)
+	}
+	t.Cleanup(func() { c.Close() })
+
+	os.Stdin = c.Tty()
+	os.Stdout = c.Tty()
+	os.Stderr = c.Tty()
+	os.Args = []string{"gh", "repo", "fork"}
+
+	donec := make(chan struct{})
+	go func() {
+		defer close(donec)
+		_, _ = c.ExpectString("Would you like to add a remote for the fork?")
+		_, _ = c.SendLine("Y")
+		_, _ = c.ExpectString("FINISHED")
+	}()
+
+	// Execute command
+	code := mainRun()
```

Comment:
Do you think that we could be executing the `gh` process from the "outside" rather than calling methods from the `main` package?

Reply:
This is doable. Is your thought that the start of an integration test suite would build the current code and then the tests would use `exec.Command` to invoke it?

Reply:
I am not sure what that buys us though considering `mainRun()` is the top level of the code.

Reply:
The approach I've described has an added benefit of testing the live `main()` function, but you are right that it would involve more steps and thus be slower and more brittle. I'm fine for now with interfacing with `mainRun()`.